hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryInterface.java

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hbase.tool;
+
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.yetus.audience.InterfaceAudience;
+
+import java.util.concurrent.ExecutorService;
+
+@InterfaceAudience.Public
+public interface CanaryInterface {
```

Comment:
What do folks think about having the interface be `Canary` and the tool be `CanaryTool` or `CanaryImpl`?

Reply:
I also thought about this. If we backport this to working branches. In my case it is 1.3. We already have other tools which calls Canary via toolRunner and if I backport this patch to 1.3 then my current code will fail since Canary will be an interface now instead of ToolRunner implementation.

Reply:
I think when you backport this patch to 1.3 you can also change the code in 1.3 to call CanaryTool instead of Canary? No big problem? CanaryInterface is not a good class name I'd say.

Reply:
> I think when you backport this patch to 1.3 you can also change the code in 1.3 to call CanaryTool instead of Canary?
Wouldn't that be an incompatible change ? In minor upgrades, if we ask downstream projects to change how they invoke public exposed classes. I am just trying to understand the compatibility challenges here. In my company, I can change my downstream project. I am not sure if other people would be fine with it. @Apache9  @busbey  Please advise.